google/cloud/pubsub/internal/tracing_batch_callback.cc

```diff
@@ -254,6 +310,10 @@ class TracingBatchCallback : public BatchCallback {
   std::mutex mu_;
   std::unordered_map<std::string, MessageSpans>
       spans_by_ack_id_;  // ABSL_GUARDED_BY(mu_)
+  std::unordered_map<
+      std::int64_t,
+      opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
```

Comment:
You might want to consider why we need to introduce a name/id for a shared_ptr, when its value seems to satisfy that role.

Reply:
Hm... originally it was because this is being called in non-otel code, but we can use the work around of the Span wrapper to solve that issue.

Reply:
Much nicer, methinks.  Note that the PR description is now largely obsolete.

Reply:
Agreed. will update.